compiler/rustc_typeck/src/check/method/mod.rs

```diff
@@ -328,21 +313,52 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             self.var_for_def(span, param)
         });
 
-        let trait_ref = ty::TraitRef::new(trait_def_id, substs);
+        let trait_ref = ty::TraitRef::new(trait_def_id, substs.clone());
```

Comment:
```suggestion
        let trait_ref = ty::TraitRef::new(trait_def_id, substs);
```
`substs` is a reference, no?